### PR TITLE
[query] Reimplement Table._same as outer join on row idx

### PR DESCRIPTION
**Pull request by tpoterba**
Is it this simple?

### CODE REVIEW

**Comment by patrick-schultz** [comment]
That looks like it should work (though some tests are failing), but rekeying by the global index might be more expensive than necessary. What about using `multi_way_zip_join`?

**Comment by tpoterba** [comment]
OK, working now.

File: hail/python/test/hail/table/test_grouped_table.py [hunk 1]
```diff
@@ -8,7 +8,8 @@
 
 
 class GroupedTableTests(unittest.TestCase):
-    def test_aggregate_by(self):
+    def \
+            test_aggregate_by(self):
```
Comment (patrick-schultz) [inline hail/python/test/hail/table/test_grouped_table.py:12]: What happened here?

Comment (tpoterba) [reply hail/python/test/hail/table/test_grouped_table.py:12]: oh, oopz

**Pull request merged**